examples/fluids/src/setupts.c

```diff
@@ -15,6 +15,11 @@
 #include "../navierstokes.h"
 #include "../qfunctions/newtonian_state.h"
 
+PetscLogEvent SmartRedis_Init;
+PetscLogEvent SmartRedis_Meta;
+PetscLogEvent SmartRedis_Train;
+PetscLogEvent TrainDataCompute;
+
```

Comment:
```suggestion
```

Reply:
I tried removing these line, but this gives a build error with undefined reference to those variables. Do you have another place in mind where they should go if not in setupts.c?

Reply:
So you should end up with:
```c
PetscLogEvent       FLUIDS_CeedOperatorApply;
PetscLogEvent       FLUIDS_CeedOperatorAssemble;
PetscLogEvent       FLUIDS_CeedOperatorAssembleDiagonal;
PetscLogEvent       FLUIDS_CeedOperatorAssemblePointBlockDiagonal;
PetscLogEvent       FLUIDS_SmartRedis_Init;
PetscLogEvent       FLUIDS_SmartRedis_Meta;
PetscLogEvent       FLUIDS_SmartRedis_Train;
PetscLogEvent       FLUIDS_TrainDataCompute;
static PetscClassId libCEED_classid, onlineTrain;

PetscErrorCode RegisterLogEvents() {
  PetscFunctionBeginUser;
  PetscCall(PetscClassIdRegister("libCEED", &libCEED_classid));
  PetscCall(PetscLogEventRegister("CeedOpApply", libCEED_classid, &FLUIDS_CeedOperatorApply));
  PetscCall(PetscLogEventRegister("CeedOpAsm", libCEED_classid, &FLUIDS_CeedOperatorAssemble));
  PetscCall(PetscLogEventRegister("CeedOpAsmD", libCEED_classid, &FLUIDS_CeedOperatorAssembleDiagonal));
  PetscCall(PetscLogEventRegister("CeedOpAsmPBD", libCEED_classid, &FLUIDS_CeedOperatorAssemblePointBlockDiagonal));

  PetscCall(PetscClassIdRegister("onlineTrain", &onlineTrain));
  PetscCall(PetscLogEventRegister("SmartRedis_Init", onlineTrain, &FLUIDS_SmartRedis_Init));
  PetscCall(PetscLogEventRegister("SmartRedis_Meta", onlineTrain, &FLUIDS_SmartRedis_Meta));
  PetscCall(PetscLogEventRegister("SmartRedis_Train", onlineTrain, &FLUIDS_SmartRedis_Train));
  PetscCall(PetscLogEventRegister("TrainDataCompute", onlineTrain, &FLUIDS_TrainDataCompute));
  PetscFunctionReturn(PETSC_SUCCESS);
}
```
in `misc.c` and:
```c
extern PetscLogEvent FLUIDS_CeedOperatorApply;
extern PetscLogEvent FLUIDS_CeedOperatorAssemble;
extern PetscLogEvent FLUIDS_CeedOperatorAssembleDiagonal;
extern PetscLogEvent FLUIDS_CeedOperatorAssemblePointBlockDiagonal;
extern PetscLogEvent FLUIDS_SmartRedis_Init;
extern PetscLogEvent FLUIDS_SmartRedis_Meta;
extern PetscLogEvent FLUIDS_SmartRedis_Train;
extern PetscLogEvent FLUIDS_TrainDataCompute;
PetscErrorCode       RegisterLogEvents();
```
in `navierstokes.h`. Note that I also added the `FLUIDS_` prefix to the event logs per the formatting used for the others, so that would also need to be updated for all the other uses if you just copy/pasted this code.

Reply:
This should be resolved with latest commits and I tested that it builds and runs.